### PR TITLE
switch traces to memphis stream

### PR DESCRIPTION
**Pull request by shohamroditimemphis**

### CODE REVIEW

File: analytics/analytics.go [hunk 1]
```diff
@@ -70,80 +86,56 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
...
+		tenantName = strings.ReplaceAll(tenantName, "-", "_") // for parsing purposes
+		if tenantName != "" && username != "" {
+			distinctId = distinctId + "-" + tenantName + "-" + username
+		}
 	}
 
-	tenantName = strings.ReplaceAll(tenantName, "-", "_") // for parsing purposes
-	if tenantName != "" && username != "" {
-		distinctId = distinctId + "-" + tenantName + "-" + username
+	conn, err := memphis.Connect(HOST, USERNAME, memphis.Password(PASSWORD), memphis.AccountId(ACCOUNT_ID))
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:105]: connect per message will lead to exhaust the cluster

File: analytics/analytics.go [hunk 2]
```diff
@@ -70,80 +86,56 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
...
-		Event:      eventName,
-		Properties: p,
-	})
-}
-
-func SendErrEvent(origin, errMsg string) {
-	distinctId := deploymentId
-	if configuration.DEV_ENV != "" {
-		distinctId = "dev"
+	err = conn.Produce("users-traces", "producer_users_traces", []byte(eventMsg), []memphis.ProducerOpt{memphis.ProducerGenUniqueSuffix()}, []memphis.ProduceOpt{})
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:135]: eventMsg is already bytes no?

File: server/memphis_helper.go [hunk 3]
```diff
@@ -1491,3 +1489,17 @@ func validatePassword(password string) error {
...
+func (s *Server) getIp() string {
+	resp, err := http.Get("https://ifconfig.me")
+	if err != nil {
+		serv.Errorf("Error get ip: %s", err.Error())
+	}
+	defer resp.Body.Close()
+
+	ip, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		serv.Errorf("Error reading response get ip body: %s", err.Error())
```
Comment (idanasulinmemphis) [inline server/memphis_helper.go:1502]: return ""

File: server/memphis_helper.go [hunk 4]
```diff
@@ -1491,3 +1489,17 @@ func validatePassword(password string) error {
 
 	return errors.New("Password must be at least 8 characters long, contain both uppercase and lowercase, and at least one number and one special character")
 }
+
+func (s *Server) getIp() string {
+	resp, err := http.Get("https://ifconfig.me")
+	if err != nil {
+		serv.Errorf("Error get ip: %s", err.Error())
```
Comment (idanasulinmemphis) [inline server/memphis_helper.go:1496]: return

File: analytics/analytics.go [hunk 5]
```diff
@@ -12,25 +12,44 @@
...
 
 	"github.com/gofrs/uuid"
-	"github.com/posthog/posthog-go"
+	"github.com/memphisdev/memphis.go"
+)
+
+const (
+	ACCOUNT_ID = 223671990
+	USERNAME   = "users_traces"
+	PASSWORD   = "usersTracesMemphis@1"
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:30]: change to traces_producer
I've already created it with the same password

File: analytics/analytics.go [hunk 6]
```diff
@@ -12,25 +12,44 @@
...
+	Properties     []EventParam `json:"properties"`
+	TimeStamp      time.Time    `json:"time_stamp"`
+	MemphisVersion string       `json:"memphis_version"`
+}
+
 var configuration = conf.GetConfig()
 var deploymentId string
 var memphisVersion string
-var AnalyticsClient posthog.Client
+var sdkConnection *memphis.Conn
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:50]: memphisConn

File: analytics/analytics.go [hunk 7]
```diff
@@ -70,80 +89,72 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
 		}
 	}
 
-	client, err := posthog.NewWithConfig(analyticsToken, posthog.Config{Endpoint: "https://app.posthog.com"})
+	sdkConnection, err = initalizeSdkConnection()
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:92]: just call connect, no need to wrap connect in a function

File: analytics/analytics.go [hunk 8]
```diff
@@ -70,80 +89,72 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
...
 	if err != nil {
-		return err
+		errMsg := fmt.Sprintf("Initalize connection failed %s ", err.Error())
+		return errors.New(errMsg)
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:95]: You can use fmt.Errorf rather than do that

Comment (idanasulinmemphis) [reply analytics/analytics.go:95]: Add the function name to the log, it is not clear what happen

File: analytics/analytics.go [hunk 9]
```diff
@@ -70,80 +89,72 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
...
+	eventMsg, err = json.Marshal(event)
+	if err != nil {
+		return
 	}
-
-	p := posthog.NewProperties()
-	for _, param := range params {
-		p.Set(param.Name, param.Value)
+	err = sdkConnection.Produce("users-traces", "producer_users_traces", eventMsg, []memphis.ProducerOpt{memphis.ProducerGenUniqueSuffix()}, []memphis.ProduceOpt{})
+	if err != nil {
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:147]: you can simply ignore errors you have 
if => return else => return

File: analytics/analytics.go [hunk 10]
```diff
@@ -70,80 +89,72 @@ func InitializeAnalytics(analyticsToken, memphisV, customDeploymentId string) er
...
-	})
+	return
 }
 
-func SendErrEvent(origin, errMsg string) {
-	distinctId := deploymentId
-	if configuration.DEV_ENV != "" {
-		distinctId = "dev"
+func initalizeSdkConnection() (*memphis.Conn, error) {
+	conn, err := memphis.Connect(HOST, USERNAME, memphis.Password(PASSWORD), memphis.AccountId(ACCOUNT_ID))
```
Comment (idanasulinmemphis) [inline analytics/analytics.go:155]: define reconnection
max reconnect = 500
reconnectWait = 1 sec

File: main.go [hunk 11]
```diff
@@ -96,7 +96,7 @@ func usage() {
 }
 
 func runMemphis(s *server.Server) {
-	err := analytics.InitializeAnalytics(s.AnalyticsToken(), s.MemphisVersion(), s.GetCustomDeploymentId())
+	err := analytics.InitializeAnalytics(s.MemphisVersion(), s.GetCustomDeploymentId())
```
Comment (idanasulinmemphis) [inline main.go:99]: again you are missing calling close function at the end of the main function

File: server/memphis_helper.go [hunk 12]
```diff
@@ -1453,3 +1451,19 @@ func (s *Server) MoveResourcesFromOldToNewDefaultAcc() error {
 	}
 	return nil
 }
+
+func (s *Server) getIp() string {
+	resp, err := http.Get("https://ifconfig.me")
+	if err != nil {
+		serv.Warnf("Error get ip: %s", err.Error())
```
Comment (idanasulinmemphis) [inline server/memphis_helper.go:1458]: add the function name to logs

File: main.go (hunk 11 above)
Comment (shohamroditimemphis) [reply main.go:99]: I already call close function as in master version

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**